aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -113,33 +113,55 @@ class KernelTable_ final {
 class DispatchTable final {
  public:
   DispatchTable(const FunctionSchema& schema)
-  : kernels_()
+  : kernels_(make_left<detail::KernelTable_, DispatchTableEntry>())
   , dispatch_strategy_(get_dispatch_strategy_(schema))
   , operator_name_(schema.name()) {}
 
   /**
    * Register a kernel in the table at some dispatch key.
    * @param dispatch_key Dispatch key to define when this kernel is selected.
-   *        If this is TensorTypeIds::undefined(), this registers a fallback
-   *        kernel that is called whenever no other kernel matches.
    * @param kernel Concrete kernel function implementation to register
    */
   void setKernel(
       TensorTypeId dispatch_key,
       const DispatchTableEntry& kernel) {
-    const bool is_fallback_kernel = (dispatch_key == TensorTypeIds::undefined());
-    AT_ASSERTM(is_fallback_kernel || dispatch_strategy_.is_valid_, "Tried to register a kernel with a dispatch key for operator schema ", operator_name_, " that doesn't have tensor arguments.");
-    kernels_.set(dispatch_key, kernel, operator_name_);
+    AT_ASSERTM(dispatch_key != TensorTypeIds::undefined());
```

Comment:
we should change these to TORCH_CHECK too (maybe in a separate PR)

Reply:
will fix in a separate diff